sim/midas/src/main/scala/midas/widgets/Assert.scala

```diff
@@ -22,23 +22,29 @@ class AssertBridgeModule(assertMessages: Seq[String])(implicit p: Parameters) ex
     val hPort = IO(HostPort(Input(UInt(numAsserts.W))))
     val resume = WireInit(false.B)
     val cycles = RegInit(0.U(64.W))
-    val asserts = hPort.hBits
+    val q = Module(new Queue(hPort.hBits.cloneType,2))
+    q.io.enq.valid := hPort.toHost.hValid
+    hPort.toHost.hReady := q.io.enq.ready
+    q.io.enq.bits := hPort.hBits
+    // We only sink tokens, so tie off the return channel
+    hPort.fromHost.hValid := true.B
+
+    val asserts = q.io.deq.bits
+
     val assertId = PriorityEncoder(asserts)
     val assertFire = asserts.orR
 
     val stallN = (!assertFire || resume)
 
-    val tFireHelper = DecoupledHelper(hPort.toHost.hValid, stallN)
+    val tFireHelper = DecoupledHelper(q.io.deq.valid, stallN)
```

Comment:
When I ran the tests for this change, I still had `dummyPredicate` as a third argument to `DecoupledHelper()` and had parens on `tFireHelper.fire()` below but I don't expect those two changes coming from upstream will break anything here.

Reply:
Yeah it should be fine.